packages/cli/src/commands/decode.ts

```diff
@@ -57,16 +57,25 @@ export async function decode({
 
     const { provider } = await resolveProvider({ action: ProviderAction.ReadProvider, quiet: true, cliSettings, chainId });
 
-    const transaction = await provider.getTransaction({
+    const tx = await provider.getTransaction({
       hash: data,
     });
 
-    inputData = transaction.input;
+    inputData = tx.input;
+
+    log('     RPC: ', provider.transport.url);
+    log('Chain ID: ', tx.chainId);
+    log(' TX hash: ', tx.hash);
+    if (tx.from) log('    From: ', tx.from);
```

Comment:
if from is not specified its probably best to say `From: undefined` rather than just not showing it.